chromium_src/ios/web/navigation/crw_wk_navigation_handler.mm

```diff
@@ -40,6 +45,18 @@ - (void)webView:(WKWebView*)webView
           // Only ever update it to false
           preferences.allowsContentJavaScript = false;
         }
+
+        const web::UserAgentType userAgentType =
+            [self userAgentForNavigationAction:action webView:webView];
+        if (userAgentType != web::UserAgentType::NONE) {
+          NSString* userAgent = brave::GetUserAgentForRequest(
+              static_cast<web::WebState*>(self.webStateImpl), userAgentType,
+              action.request);
+          if (userAgent && webView.customUserAgent != userAgent) {
```

Comment:
Do we have any testing framework on iOS to test the change?

Reply:
Not at the moment, but this is essentially the same logic thats in place atm in Chromium: https://source.chromium.org/chromium/chromium/src/+/main:ios/web/navigation/crw_wk_navigation_handler.mm;l=240

Reply:
I did just notice though that `webView.customUserAgent != userAgent` needs to actually be `![webView.customUserAgent isEqualToString:userAgent]` (darn Obj-C…), pushing a fix for that momentarily